File: leetcode/js/range-addition-ii.js
/**
 * @param {number} m
 * @param {number} n
 * @param {number[][]} ops
 * @return {number}
 */
var maxCount = function (m, n, ops) {
  let minRow = m;
  let minCol = n;

  for (const [r, c] of ops) {
    minRow = Math.min(minRow, r);
    minCol = Math.min(minCol, c);
  }

  return minRow * minCol;
};

File: leetcode/rust/range-addition-ii.rs
impl Solution {
    pub fn max_count(m: i32, n: i32, ops: Vec<Vec<i32>>) -> i32 {
        let mut min_row = m;
        let mut min_col = n;

        for op in ops {
            min_row = min_row.min(op[0]);
            min_col = min_col.min(op[1]);
        }

        min_row * min_col
    }
}
